src/Kestrel.Core/Internal/Http2/Http2Connection.cs

```diff
@@ -87,8 +87,11 @@ public Http2Connection(Http2ConnectionContext context)
         {
             _context = context;
             _frameWriter = new Http2FrameWriter(context.Transport.Output, context.ConnectionContext, _outputFlowControl, this, context.ConnectionId, context.ServiceContext.Log);
-            _hpackDecoder = new HPackDecoder((int)_serverSettings.HeaderTableSize);
             _serverSettings.MaxConcurrentStreams = (uint)context.ServiceContext.ServerOptions.Limits.Http2.MaxStreamsPerConnection;
+            _serverSettings.MaxFrameSize = (uint)context.ServiceContext.ServerOptions.Limits.Http2.MaxFrameSize;
+            _serverSettings.HeaderTableSize = (uint)context.ServiceContext.ServerOptions.Limits.Http2.HeaderTableSize;
+            _hpackDecoder = new HPackDecoder((int)_serverSettings.HeaderTableSize);
```

Comment:
There is no corresponding logic required to handle the HeaderTableSize of the _clientSettings since we are not currently doing header compression in our HPackEncoder. Our HPackEncoder currently does a direct literal encoding without using a compression table or huffman encoding; I should file an issue for that.